perfkitbenchmarker/linux_benchmarks/aws_dynamodb_ycsb_benchmark.py

```diff
@@ -0,0 +1,161 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Run YCSB benchmark against AWS DynamoDB.
+This benchmark does not provision VMs for the corresponding DynamboDB database.
+The only VM group is client group that sends requests to specifiedDB.
+Before running this benchmark, you have to manually create `usertable` with primaryKey 'user02'.
+TODO: fix _Exists logic.
+TODO: add DAX option.
+TODO: add global table option
+"""
+
+import posixpath
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.linux_packages import ycsb
+from perfkitbenchmarker.providers.aws import aws_dynamodb
+
+BENCHMARK_NAME = 'aws_dynamodb_ycsb'
+BENCHMARK_CONFIG = """
+aws_dynamodb_ycsb:
+  description: >
+      Run YCSB against AWS DynamoDB.
+      Configure the number of VMs via --ycsb_client_vms.
+  vm_groups:
+    default:
+      vm_spec: *default_single_core
+      vm_count: 1"""
+
+YCSB_BINDING_LIB_DIR = posixpath.join(ycsb.YCSB_DIR, 'dynamodb-binding', 'lib')
+AWS_CREDENTIAL_DIR = '/tmp/AWSCredentials.properties'
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('aws_dynamodb_ycsb_awscredentials_properties',
+                    './AWSCredentials.properties',
+                    'The AWS credential location.')
+flags.DEFINE_string('aws_dynamodb_ycsb_dynamodb_primarykey',
+                    'user01',
+                    'The primaryKey of table.')
+flags.DEFINE_string('aws_dynamodb_ycsb_dynamodb_endpoint',
+                    None,
+                    'The aws dynamodb endpoint to connect to.')
+flags.DEFINE_string('aws_dynamodb_ycsb_requestdistribution',
+                    'uniform',
+                    'Type of request distribution.')
+flags.DEFINE_string('aws_dynamodb_ycsb_readproportion',
+                    '0.5',
+                    'The Read Proportion, default is 0.5 in workloada and 0.95 in YCSB.')
+flags.DEFINE_string('aws_dynamodb_ycsb_updateproportion',
+                    '0.5',
+                    'The Update Proportion, default is 0.5 in workloada and 0.05 in YCSB')
+flags.DEFINE_string('aws_dynamodb_ycsb_table',
+                    'usertable',
+                    'The Table flag determines which table to run on.')
+flags.DEFINE_string('aws_dynamodb_ycsb_consistentReads',
+                    'false',
+                    'Consistent reads cost 2x Eventual reads. False is default and eventual')
+flags.DEFINE_string('aws_dynamodb_ycsb_capacity',
```

Comment:
this is dynamodb specific completely unrelated to ycsb

Reply:
this flag is defined in the dynamodb ycsb config file: https://github.com/brianfrankcooper/YCSB/blob/master/dynamodb/conf/dynamodb.properties

Reply:
Can you be more specific, besides "workload size has implications on provisioned read and 
capacity", i didn't see any other place mentioning capacity in either property file nor the dynamodb client implementation https://github.com/brianfrankcooper/YCSB/blob/master/dynamodb/src/main/java/com/yahoo/ycsb/db/DynamoDBClient.java.

Reply:
moved capacity flag to aws_dyanamodb.py since it was not in ycsb.